### ISSUE TITLE
Initial build

### ISSUE THREAD

**Issue by mc738** [mc738/dmap#7]
Create initial build and functionality. Once this is in place future work should be managed with pull requests.

**Comment by mc738** [comment]
This was left as a reminder, a few this:

* Flatten removed the need to traverse directories to compare diffs
* if filemap_1 is bigger than filemap_2 BUT has different items, in theory it could end up smaller and items could be missed when they are compared as equal size, however this is not the case (so far...) in practice.
    * Does this mean the diff algorithm can be simplified further?

**Comment by mc738** [comment]
Complete
